### PR TITLE
Fix for SubmodelComp indexing bug.

### LINKED ISSUE #3200: Submodel outputs might not be set correctly
### Description

Might be a similar fix to #3192? I'm not certain of the inner workings. Seems it might be related to the names of the subsystems or the order? I've attached an a test showing the desired behavior and the failing behavior.

### Example

```
import unittest
import numpy as np

import openmdao.api as om
from openmdao.utils.assert_utils import assert_near_equal


def build_submodel(subsystem_name):
    p = om.Problem()
    supmodel = om.Group()
    supmodel.add_subsystem('supComp', om.ExecComp('diameter = r * theta'),
                            promotes_inputs=['*'],
                            promotes_outputs=['*', ('diameter', 'aircraft:fuselage:diameter')])

    subprob1 = om.Problem()
    submodel1 = subprob1.model.add_subsystem('submodel1', om.Group(), promotes=['*'])

    submodel1.add_subsystem(subsystem_name, om.ExecComp('x = diameter * 2 * r * theta'), promotes=['*', ('diameter', 'aircraft:fuselage:diameter')])
    submodel1.add_subsystem('b', om.ExecComp('y = mass * donkey_kong'), promotes=['*', ('mass', 'dynamic:mission:mass'), ('donkey_kong', 'aircraft:engine:donkey_kong')])

    p.model.add_subsystem('supModel', supmodel, promotes_inputs=['*'],
                            promotes_outputs=['*'])


    submodel = om.SubmodelComp(problem=subprob1, inputs=['*'], outputs=['*'])
    p.model.add_subsystem('sub1', submodel,
                            promotes_inputs=['*'],
                            promotes_outputs=['*'])

    p.model.set_input_defaults('r', 1.25)
    p.model.set_input_defaults('theta', np.pi)

    p.setup(force_alloc_complex=True)

    p.set_val('r', 1.25)
    p.set_val('theta', 0.5)
    p.set_val('dynamic:mission:mass', 2.0)
    p.set_val('aircraft:engine:donkey_kong', 3.0)
    p.set_val('aircraft:fuselage:diameter', 3.5)

    return p



class TestSubModelBug(unittest.TestCase):
    
    def test_submodel_bug(self):
        p = build_submodel(subsystem_name='a')

        p.run_model()

        assert_near_equal(p.get_val('y'), 2.0 * 3.0)
    
    
    def test_submodel_bug_fails(self):
        p = build_submodel(subsystem_name='c')

        p.run_model()

        assert_near_equal(p.get_val('y'), 2.0 * 3.0)


if __name__ == '__main__':
    unittest.main()
```

### OpenMDAO Version

3.31.2-dev

### Relevant environment information

_No response_

### PR DESCRIPTION
**Pull request by naylor-b**
### Summary

SubmodelComp outputs were not being properly copied to the outer model in some cases.

For now, variables exposed as inputs or outputs of a SubmodelComp are not allowed to be distributed.

### Related Issues

- Resolves #3200

### Backwards incompatibilities

The default value of the `do_coloring` option for SubmodelComp has been changed to False.

### New Dependencies

None

### CODE REVIEW

File: openmdao/components/tests/test_submodel_comp.py [hunk 1]
```diff
@@ -452,15 +454,83 @@ def test_submodel_comp(self):
...
+                                                      do_coloring=True))
+        par.add_subsystem('subprob2', om.SubmodelComp(problem=om.Problem(model=FanInGrouped()),
+                                                      inputs=['*'], outputs=['c3.y'],
+                                                      do_coloring=True))
 
         p.setup(force_alloc_complex=True)
         p.run_model()
         assert_check_partials(p.check_partials(method='cs', out_stream=None))
 
+    # @unittest.skip("Unskip this after vars under ParallelGroups work with SubmodelComp")
```
Comment (robfalck) [inline openmdao/components/tests/test_submodel_comp.py:468]: Should this be commented?

File: openmdao/core/system.py [hunk 2]
```diff
@@ -3609,7 +3609,7 @@ def _update_response_meta(self, meta, get_size=False, use_prom_ivc=False):
             key = src_name
 
         meta['source'] = src_name
-        meta['distributed'] = \
+        meta['distributed'] = dist = \
```
Comment (robfalck) [inline openmdao/core/system.py:3612]: dist is never used

**Review by robfalck — CHANGES_REQUESTED** [review]

File: openmdao/core/system.py (hunk 2 above)
Comment (crecine) [reply openmdao/core/system.py:3612]: it's used on line 3629 as part of the `if`

**Review by johnjasa — APPROVED** [review]

**Comment by johnjasa** [comment]
Thanks for this fix and cleanup, Bret! We really appreciate it.

File: openmdao/components/tests/test_submodel_comp.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/components/tests/test_submodel_comp.py]: I just removed it.  At one point when I was trying to get distrib vars to work, both parallel group vars and distrib vars were broken, but when I backed out all of those changes, the parallel vars worked, so right now the only thing that doesn't work is the distrib vars.  Actually Ken found one more issue (no support for unit conversions between inner and outer model), but I'll fix that in a follow-on PR.

**Review by robfalck — APPROVED** [review]

**Pull request merged**